lib/tokens/token.js

```diff
@@ -40,28 +40,34 @@ module.exports = function (log, random, P, hkdf, Bundle, error) {
     this.algorithm = 'sha256'
     this.uid = details.uid || null
     this.lifetime = details.lifetime || Infinity
-    this.createdAt = optionallyOverrideCreatedAt(details.createdAt)
+    this.createdAt = details.createdAt || 0
   }
 
   function optionallyOverrideCreatedAt (timestamp) {
     var now = Date.now()
-
-    if (! config.isProduction && timestamp >= 0 && timestamp < now) {
-      // In the wild, all tokens should have a fresh createdAt timestamp.
-      // For testing purposes only, allow createdAt to be overridden.
-      return timestamp
+    if (typeof timestamp !== 'number' || isNaN(timestamp)) {
+      return now
     }
-
-    return now
+    if (timestamp < 0 || timestamp > now) {
+      return now
+    }
+    if (config.isProduction) {
```

Comment:
Maybe promote this test up to first place so that the production path is the fastest one?